include/pressiodemoapps/impl/euler_2d_initial_condition.hpp

```diff
@@ -319,19 +325,53 @@ void riemann2dIC2(state_type & state,
   const auto & x= meshObj.viewX();
   const auto & y= meshObj.viewY();
   std::array<scalar_type, numDofPerCell> prim = {};
+
+  // some checks
+  if (topRightPressure <= botLeftPressure) {
+    throw std::runtime_error("INVALID: riemannTopRightPressure <= riemannBotLeftPressure");
+  }
+
+  // compute remaining quantities
+  // compatibility equations
```

Comment:
can you please add the reference(s) where all this is coming from?

Reply:
I included the reference in the comments above the function, though I think your comment was made on an old version of the file (commit `ed9670e`) and so the paper link doesn't appear there. Should be there if you select the later commits.